### DISCUSSION TITLE
ssh-keygen -i can't import private PKCS8 keys

### DISCUSSION THREAD

**Discussion by karamellpelle** [openssl/openssl#24114] [Q&A] [answered]
`ssh -V`: `OpenSSH_9.7p1, OpenSSL 3.2.1 30 Jan 2024` (_mingw64_ on Windows 11 since _WSL_ has too low version --- I haven't had access to Linux ATM).

According to [documentation](https://man.archlinux.org/man/ssh-keygen.1#i) for `ssh-keygen -i`:  

> This option will read an unencrypted **private** (or public) key file in the format specified
> by the -m option and print an OpenSSH compatible **private** (or public) key to stdout.
> This option allows importing keys from other software, including several commercial SSH implementations. 
> The default import format is “RFC4716”.

However, I can' import a private key generated from _OpenSSL_

~~~zsh
$ openssl genpkey -algorithm rsa  -out rsa-priv.pem # create unencrypted PKCS8 (the default output format) private key
$ ssh-keygen -i -m PKCS8 -f rsa-priv.pem
do_convert_from_pkcs8: rsa-priv.pem is not a recognised public key format
~~~

However, importing the public part of the SSL keypair works:

~~~zsh
$ openssl pkey -in rsa-priv.pem -pubout -out rsa-pub.pem
$ ssh-keygen -i -f rsa-pub.pem -m PKCS8
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC6wjzm5Rx5W78nnHrY8bzpUtNvYf0/W7n3KLLa3a+WKI7noOw6fan9ZKTAZvj7cm+6/eTS9YZwMJkKBsuk6Blw0ykih7FXP5VJ92ue9KH7SEKOJ7NG2LH14O9D6FSciHEcKXhpcysieD515sLX2W8NNFh3IzSWdjw8aADL8kdqqAnrv1et22RTmPmEfgFdZ66s26kyHjROxzjenMZiZCX/1unFxogdIxgsDlt2fE4SQtE9/ngM+AslKJDWLNxGHorVEXlIAdUdKjPkiLfO0a0DsPNpX+oABtnD1RDdxxNCFhPrILP+HnlqRtSu8tJcLxmk8va4StHUBBQXITDP6HDJ
~~~

Am I doing something wrong, or is this a bug?

---
The _man_ version is _9.7p1-1_, but I don't think that's the problem here (I've read the changelog).

**Comment by t8m** [comment, marked as answer]
It might be a bug, but I do not think this is an OpenSSL bug. It could be a bug in OpenSSH.

**Reply by karamellpelle** [reply]
This question/bug report was (obviously?) made for _OpenSSH_, but I did a mistake here and mixed SSL with SSH 😂. I'm sorry. You can remove this post from me, and I can correctly post it to **OpenSSH** instead!

I don't think there is something wrong with OpenSSL!

Thank you so much for a fast answer 🙂
